dbt/include/redshift/macros/adapters.sql

```diff
@@ -43,6 +43,24 @@
 
   {{ sql_header if sql_header is not none }}
 
+  {%- if config.get('constraints_enabled', False) %}
+
+  create {% if temporary -%}temporary{%- endif %} table
+    {{ relation.include(database=(not temporary), schema=(not temporary)) }}
+    {{ get_columns_spec_ddl() }}
```

Comment:
```suggestion
    {{ get_columns_spec_ddl() }}
    {{ get_assert_columns_equivalent(sql) }}
```

This creates a validation check that columns in a sql statement match the schema config